### PR TITLE
Feature/fix field naming

### PR DESCRIPTION
**Pull request by chasegiunta**
Since we're allowing either `name` or `label` to be passed into`Field`, whichever is passed needs to be the key used in `Form.fields`.

Before, the field was being registered with `field: this.name` but since `name` wasn't required, input on the field would break.

Down the road we may need to account for duplicate names, but assuming someone is using say, and "Email" field twice, we can assume for now that they are smart enough to pass their own `name` prop.

### CODE REVIEW

File: src/Field.vue [hunk 1]
```diff
@@ -81,6 +84,17 @@ export default {
...
     },
+    fieldName() {
+      if (this.name !== undefined) {
+        return this.name
+      } else if (this.label !== undefined) {
+        return this.label
+      } else {
+        // This auto trims the contents for us, but seems dirty.
+        // Maybe set a reference on the default slot instead and grab $el.innerHtml?
+        return this.$slots.default[0].text
```
Comment (chasegiunta) [inline src/Field.vue]: @jakedohm Either way is dirty, and I don't know which I hate more. Thoughts?

File: src/Field.vue [hunk 2]
```diff
@@ -81,6 +84,17 @@ export default {
     fieldLabel() {
       return this.label || this.name
     },
+    fieldName() {
```
Comment (chasegiunta) [inline src/Field.vue]: @jakedohm Is there a cleaner way to do this than if/elseif,/else? Thought I could `!== null` here but apparently not (or, maybe I could if we set the prop defaults to `null`)

File: src/Field.vue (hunk 1 above)
Comment (jakedohm) [reply src/Field.vue]: This looks good, that's how I was doing it!

File: src/Field.vue (hunk 2 above)
Comment (jakedohm) [reply src/Field.vue]: The way you're doing it looks perfect!

**Pull request merged**